src/InstrumentationEngine/ProfilerManagerWrapper.cpp

```diff
@@ -0,0 +1,237 @@
+/*
+ * Inject Instrumentation Method ClassId
+ */
+HRESULT CProfilerManagerWrapper::LogMessage(_In_ const WCHAR* wszMessage)
+{
+    HRESULT hr = S_OK;
+    WCHAR wszBuffer[LogEntryMaxSize] = { 0 };
+    tstring tsMessage = wszMessage;
+    tstring tsEscapedMessage;
+    for (tstring::iterator it = tsMessage.begin(); it < tsMessage.end(); it++)
+    {
+        // Escape any existing format strings to avoid conflicts with the FORMATTABLE_PREFIX
+        if (*it == L'%')
+        {
+            tsEscapedMessage.push_back(*it);
+        }
+        tsEscapedMessage.push_back(*it);
+    }
+    IfFailRetErrno(wcscat_s(wszBuffer, FORMATTABLE_PREFIX));
+    IfFailRetErrno(wcscat_s(wszBuffer, tsEscapedMessage.c_str()));
+
+    return m_pProfilerManager->LogMessageEx(wszBuffer, m_wszInstrumentationMethodGuid.c_str());
+}
+
+/*
+ * Inject Instrumentation Method ClassId
+ */
+HRESULT CProfilerManagerWrapper::LogError(_In_ const WCHAR* wszError)
+{
+    HRESULT hr = S_OK;
+    WCHAR wszBuffer[LogEntryMaxSize] = { 0 };
+    tstring tsMessage = wszError;
+    tstring tsEscapedMessage;
+    for (tstring::iterator it = tsMessage.begin(); it < tsMessage.end(); it++)
+    {
+        // Escape any existing format strings to avoid conflicts with the FORMATTABLE_PREFIX
+        if (*it == L'%')
+        {
+            tsEscapedMessage.push_back(*it);
+        }
+
+        tsEscapedMessage.push_back(*it);
+    }
+    IfFailRetErrno(wcscat_s(wszBuffer, FORMATTABLE_PREFIX));
+    IfFailRetErrno(wcscat_s(wszBuffer, tsEscapedMessage.c_str()));
+
+    return m_pProfilerManager->LogErrorEx(wszBuffer, m_wszInstrumentationMethodGuid.c_str());
+}
+
+/*
+ * Inject Instrumentation Method ClassId
+ */
+HRESULT CProfilerManagerWrapper::LogDumpMessage(_In_ const WCHAR* wszMessage)
+{
+    HRESULT hr = S_OK;
+    WCHAR wszBuffer[LogEntryMaxSize] = { 0 };
+    tstring tsMessage = wszMessage;
+    tstring tsEscapedMessage;
+    for (tstring::iterator it = tsMessage.begin(); it < tsMessage.end(); it++)
+    {
+        // Escape any existing format strings to avoid conflicts with the FORMATTABLE_PREFIX
+        if (*it == L'%')
+        {
+            tsEscapedMessage.push_back(*it);
+        }
+
+        tsEscapedMessage.push_back(*it);
+    }
+    IfFailRetErrno(wcscat_s(wszBuffer, FORMATTABLE_PREFIX));
+    IfFailRetErrno(wcscat_s(wszBuffer, tsEscapedMessage.c_str()));
+    return m_pProfilerManager->LogDumpMessageEx(wszBuffer, m_wszInstrumentationMethodGuid.c_str());
+
+}
+
+HRESULT CProfilerManagerWrapper::EnableDiagnosticLogToDebugPort(_In_ BOOL enable)
+{
+    return m_pProfilerManager->EnableDiagnosticLogToDebugPort(enable);
+}
+
+HRESULT CProfilerManagerWrapper::GetLoggingFlags(_Out_ LoggingFlags* pLoggingFlags)
+{
+    return m_pProfilerManager->GetLoggingFlags(pLoggingFlags);
+}
+
+HRESULT CProfilerManagerWrapper::SetLoggingFlags(_In_ LoggingFlags loggingFlags)
+{
+    return m_pProfilerManager->SetLoggingFlags(loggingFlags);
+}
+
+#undef PREPEND_FORMATTABLE_PREFIX
```

Comment:
This wasn't defined, so this is doing nothing. Did you mean "#undef FORMATTABLE_PREFIX"?